### PR TITLE
mwcu_printutil: fix printing containers with custom allocators

### PR DESCRIPTION
**Pull request by ddragan-bloomberg**

### CODE REVIEW

File: src/groups/mwc/mwcu/mwcu_printutil.h [hunk 1]
```diff
@@ -738,72 +748,71 @@ mwcu::operator<<(bsl::ostream&                      stream,
...
     if (!printer.obj().empty()) {
-        typedef
-            typename bsl::unordered_set<TYPE1, TYPE2, TYPE3>::const_iterator
-                Iter;
-        Iter    begin = printer.obj().begin();
-        Iter    end   = printer.obj().end();
+        typedef typename bsl::unordered_set<KEY, HASH, KEY_EQ, ALLOC>::
+            const_iterator Iter;
+        Iter               begin = printer.obj().begin();
+        Iter               end   = printer.obj().end();
```
Comment (678098) [inline src/groups/mwc/mwcu/mwcu_printutil.h:810]:
```suggestion
        const Iter         begin = printer.obj().begin();
        const Iter         end   = printer.obj().end();
```

File: src/groups/mwc/mwcu/mwcu_printutil.h [hunk 2]
```diff
@@ -812,25 +821,28 @@ inline bsl::ostream& mwcu::operator<<(
...
 
-    typedef
-        typename bsl::unordered_map<TYPE1, TYPE2, TYPE3, TYPE4>::const_iterator
-            Iter;
-    Iter    begin = printer.obj().begin();
-    Iter    end   = printer.obj().end();
+    typedef typename bsl::unordered_map<KEY, VAL, HASH, KEY_EQ, ALLOC>::
+        const_iterator Iter;
+    Iter               begin = printer.obj().begin();
+    Iter               end   = printer.obj().end();
```
Comment (678098) [inline src/groups/mwc/mwcu/mwcu_printutil.h:838]:
```suggestion
    const Iter         begin = printer.obj().begin();
    const Iter         end   = printer.obj().end();
```

**Review by 678098 — APPROVED** [review]
Good, feel free to merge or apply the small suggestions

File: src/groups/mwc/mwcu/mwcu_printutil.h (hunk 1 above)
Comment (ddragan-bloomberg) [reply src/groups/mwc/mwcu/mwcu_printutil.h:810]: I would rather not introduce logic changes in this PR

Comment (ddragan-bloomberg) [reply src/groups/mwc/mwcu/mwcu_printutil.h:810]: unless there is a reason to, like a compiler warning or something

Comment (678098) [reply src/groups/mwc/mwcu/mwcu_printutil.h:810]: Let's merge it like this then

**Pull request merged**